src/features/vote/ActiveRequestsForm.tsx

```diff
@@ -61,12 +67,30 @@ const ActiveRequestsForm: FC<Props> = ({
     state: { address, network, signer },
   } = useContext(OnboardContext);
   const { votingContract } = useVotingContract(signer, isConnected, network);
-  // const { data } = useVotesCommittedEvents(votingContract, address);
 
   const [tableValues, setTableValues] = useState<TableValue[]>([]);
 
   const { data: roundId } = useCurrentRoundId();
   const { isOpen, open, close, modalRef } = useModal();
+  const [canReveal, setCanReveal] = useState(false);
+
+  useEffect(() => {
+    if (encryptedVotes.length && votePhase === "Reveal") {
+      if (revealedVotes.length) {
+        revealedVotes.forEach((el) => {
+          const findRevealedVote = encryptedVotes.find(
+            (x) => x.identifier === el.identifier
```

Comment:
Will this cause false-positives (and thus cause set reveal to be false when it should be true)? What happens if there are multiple votes in a round with the same identifier, but different ancillary data and/or timestamp?

Don't we need to compare ancillary data and timestamp too?

Reply:
I think the data filters based on the current round, but I could verify that with a bigger data set. If so, you would just need to add those comparisons like you said.

Reply:
Yep, I'm saying I think it needs to compare ancillaryData and timestamp in addition to the identifier. And yeah, agreed that roundId comparison shouldn't be necessary.

Reply:
Sounds good! I'll add it in.